index.ts

```diff
@@ -400,16 +407,16 @@ function getBranch(tree: MapResult, path?: string): MapResultKey {
         return tree;
     }
 
-	let branch: MapResultKey = tree;
     for (const fieldName of path.split('.')) {
-        if (!branch[fieldName]) {
+        const branch = tree[fieldName];
```

Comment:
The same here - what is the reson?

Reply:
Same as before. See [TS playground](https://www.typescriptlang.org/play?#code/FAFwngDgpgBAsgQwgJSgZwK4BsQGkpgwC8MAZglmrAD7xKqY4DcwUAHhAPYBOIM40OinTY+JAN4wA2gGsCALhhoQ3AJYA7AOYBdRYmGM8BGAF8WwUhnUBjEKs7qY1hFQCCAChVQoe+iJwANGSqUFgAJgByCAC2PkoqGpoAlIoAbpyqYTDiwDAwqqQwntzeUqQh4VGx2knZuXn8JbAkXlBlFZExUNoseSbA-RZWtvaOzlQAQsXevgaiQeWhnbGKympaKTDpmXV51g7KMABG3Ag2ABbEjaWLlV099QVFJ2fW57U5DdfNx6cXvaYBkA) for an explanation.